src/api/c/libspeechd.c

```diff
@@ -114,53 +114,65 @@ char *strndup(const char *s, size_t n)
 }
 #endif /* HAVE_STRNDUP */
 
-#ifndef HAVE_GETLINE
-#define BUFFER_LEN 256
-ssize_t getline(char **lineptr, size_t * n, FILE * f)
+#define SPD_REPLY_BUF_SIZE 65536
+
+/** Read from the connection's buffer or socket until a newline is
+    read.  Return a pointer to the beginning of the data, and the
+    length of the data (including newline).  The returned value will
+    not be null-terminated, and will include the newline.  Note that
+    the returned pointer will only be valid until the next call to
+    get_line.
+
+    If, after SPD_REPLY_BUF_SIZE bytes, or in the event of a read
+    error, there is no newline, return NULL.
```

Comment:
"there is no newline" should be before  the "or in the event", otherwise the sentence looks odd :)